boa_engine/src/object/builtins/jstypedarray.rs

```diff
@@ -273,6 +273,84 @@ impl JsTypedArray {
         )
     }
 
+    /// Calls `TypedArray.prototype.findIndex()`.
+    #[inline]
+    pub fn find_index(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<Option<usize>> {
+        let index = BuiltinTypedArray::find_index(
+            &self.inner.clone().into(),
+            &[predicate.into(), this_arg.into_or_undefined()],
+            context,
+        )?
+        .as_number()
+        .expect("TypedArray.prototype.fineIndex() should always return number");
+
+        #[allow(clippy::float_cmp)]
+        if index == -1.0 {
+            Ok(None)
+        } else {
+            Ok(Some(index as usize))
+        }
+    }
+
+    /// Calls `TypedArray.prototype.findLast()`.
+    #[inline]
+    pub fn find_last(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        BuiltinTypedArray::find_last(
+            &self.inner.clone().into(),
+            &[predicate.into(), this_arg.into_or_undefined()],
+            context,
+        )
+    }
+
+    /// Calls `TypedArray.prototype.findLastIndex()`.
+    #[inline]
+    pub fn find_last_index(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<Option<usize>> {
+        let index = BuiltinTypedArray::find_last_index(
+            &self.inner.clone().into(),
+            &[predicate.into(), this_arg.into_or_undefined()],
+            context,
+        )?
+        .as_number()
+        .expect("TypedArray.prototype.findLastIndex() should always return number");
+
+        #[allow(clippy::float_cmp)]
+        if index == -1.0 {
+            Ok(None)
+        } else {
+            Ok(Some(index as usize))
```

Comment:
Should probably use `JsValue::to_index` just in case.

Reply:
unfortunately, `JsValue::to_index` returns a `u64` and it can't be -1

Reply:
I meant after checking that `index >= 0.0`, but on second thought I think it would also complicate the implementation slightly, so I think this is fine.